### PR TITLE
Merge in btcd commit f68cd7422dd5d0e0d6002647305c1fd663aee77d

### PR DESCRIPTION
**Pull request by cjepson**
Merges in btcd commit f68cd7422dd5d0e0d6002647305c1fd663aee77d.

### CODE REVIEW

**Comment by alexlyp** [comment]
tested on simnet and everything seems to be working as expected.

File: wire/msgtx.go [hunk 1]
```diff
@@ -1701,19 +1662,16 @@ func writeTxInWitnessValueSigning(w io.Writer, pver uint32, version int32,
 // readTxOut reads the next sequence of bytes from r as a transaction output
 // (TxOut).
 func readTxOut(r io.Reader, pver uint32, version int32, to *TxOut) error {
-	var buf8 [8]byte
-	_, err := io.ReadFull(r, buf8[:])
+	value, err := binarySerializer.Uint64(r, littleEndian)
```
Comment (cjepson) [inline wire/msgtx.go]: I had thought that readElement caused an extra alloc, or at least spent an extra check for which type was which before reading into it. So, all readElements were changed to use the serializer instead.

File: wire/msgtx.go [hunk 2]
```diff
@@ -1098,14 +1101,7 @@ func (msg *MsgTx) LegacyBtcEncode(w io.Writer, pver uint32) error {
 		}
 	}
 
-	binary.LittleEndian.PutUint32(buf[:], msg.LockTime)
-	_, err = w.Write(buf[:])
-	if err != nil {
-		return err
-	}
-
-	binary.LittleEndian.PutUint32(buf[:], msg.Expiry)
```
Comment (cjepson) [inline wire/msgtx.go]: This was actually erroneously present, as legacy Bitcoin transactions do not have expiry.  The legacy functions aren't tested, and I may just remove them in the future.

Comment (davecgh) [inline wire/msgtx.go]: Alright, that makes sense.

File: wire/msgtx.go (hunk 1 above)
Comment (davecgh) [inline wire/msgtx.go]: Sounds good.

**Pull request merged**